nft/src/lib.rs

```diff
@@ -873,25 +887,31 @@ pub mod pallet {
 				ensure!(nft.state.is_syncing, Error::<T>::NFTAlreadySynced);
 
 				SecretNftsShardsCount::<T>::try_mutate(nft_id, |maybe_shards| -> DispatchResult {
+					let (cluster_id, enclave_id) = Self::get_enclave(&who).unwrap();
```

Comment:
We must never use unwrap() in pallet code. We can only use it in tests, benchmarks, or function only used in tests and benchmarks